src/components/DataDimension/CalculationModal.js

```diff
@@ -128,97 +114,94 @@ const CalculationModal = ({
-                                                },
+                                                getNewDataItem(value, label),
                                             ])
                                         )
                                     }}
                                 />
                                 <MathOperatorSelector
-                                    onSelect={({ value }) => {
+                                    onSelect={({ id }) => {
                                         setValidationOutput()
+                                        const newItem = getNewOperatorItem(id)
+                                        if (newItem.type === TYPE_INPUT) {
+                                            setFocusId(newItem.id)
+                                        }
                                         setExpressionArray((prevArray) =>
-                                            prevArray.concat([
-                                                {
-                                                    label: value,
-                                                    value,
-                                                },
-                                            ])
+                                            prevArray.concat([newItem])
                                         )
                                     }}
                                 />
                             </div>
                             <div className="right-section">
                                 <FormulaField
-                                    expression={expressionArray}
-                                    onPartSelection={onPartSelection}
-                                    selectedPart={selectedPart}
+                                    items={expressionArray}
+                                    selectedId={selectedId}
+                                    focusId={focusId}
+                                    onChange={setItemValue}
+                                    onClick={selectItem}
+                                    onDoubleClick={removeItem}
                                 />
-                                <p>
-                                    {/* TODO: Remove, for testing only */}
-                                    {parseArrayToExpression(expressionArray)}
-                                </p>
-                                <div className="actions-wrapper">
-                                    <Button small onClick={validateExpression}>
-                                        {/* TODO: add loading state to button? */}
-                                        {i18n.t('Check formula')}
-                                    </Button>
-                                    {(selectedPart || selectedPart === 0) && (
-                                        <div className={'remove-button'}>
+                                <div className="leftpad">
```

Comment:
Rename to something descriptive of what it is (not what it does), seems to be the pattern we're going for in general. Also, should use `kebab-case`.

Reply:
Fixed.